### PR TITLE
feat(table): table component implementation

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**Table**
Shortcut ticket: [sh-208406](https://app.shortcut.com/sci-design-system/story/208406/table-component)

Implemented with &lt;table&gt; under the hood.

## Checklist

- [x] Default Story in Storybook
- [ ] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
Looking good, @masoudmanson ! See my comments in chromatic regarding the sort icon placement.

**Comment by masoudmanson** [comment]
> Looking good, @masoudmanson ! See my comments in chromatic regarding the sort icon placement.

Thank you @clarsen-czi 
I updated the chevron icon.

**Review by clarsen-czi — CHANGES_REQUESTED** [review]
Looks great @masoudmanson ! I just have a small change regarding what component is displayed in CellComponent, but the CellHeader is perfect!

File: src/core/Table/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            secondaryText="2,344,000"
+            tertiaryText="12.5%"
+            horizontalAlign="right"
+            shouldShowTooltipOnHover={false}
+          />
+        </TableRow>
+        <TableRow
+          disabled
+          shouldShowTooltipOnHover
+          // hover
```
Comment (bethbertozzi) [inline src/core/Table/index.stories.tsx:66]: this can be removed i think

File: src/core/Table/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import Meta, { Test as TestStory } from "./index.stories";
+
+// Returns a component that already contain all decorators from story level, meta level and global level.
+const Test = composeStory(TestStory, Meta);
+
+describe("<Table />", () => {
+  generateSnapshots(snapshotTestStoryFile);
+
+  it("renders table component", () => {
+    render(<Test />);
```
Comment (bethbertozzi) [inline src/core/Table/index.test.tsx:15]: add tests to ensure children rendered

**Review by bethbertozzi — CHANGES_REQUESTED** [review]

**Review by bethbertozzi — APPROVED** [review]
Thanks Masoud, looks good!

File: src/core/TableRow/index.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ const TableRow = forwardRef<HTMLTableRowElement, RowProps>(
       tooltipProps,
       tooltipText = false,
       tooltipSubtitle,
-      hover,
+      hover = true,
```
Comment (mmmmmmaya) [inline src/core/TableRow/index.tsx:23]: so the row tooltip will show itself unless it's explicitly told not to? that seems a little backwards from what I would expect...

Comment (masoudmanson) [reply src/core/TableRow/index.tsx:23]: The `hover` prop determines if the row is hoverable or not (The background-color of a hoverable row changes on hover)
On the other hand, the condition to show a row tooltip is `if (shouldShowTooltipOnHover && hover)`. 
So we won't show the tooltip if the row is not hoverable.

**Review by mmmmmmaya — APPROVED** [review]

**Pull request merged**